sympy/core/__init__.py

```diff
@@ -33,3 +33,54 @@
 EulerGamma = S.EulerGamma
 GoldenRatio = S.GoldenRatio
 TribonacciConstant = S.TribonacciConstant
+
+__all__ = [
+'sympify', 'SympifyError',
+
+'cacheit',
+
+'Basic', 'Atom', 'preorder_traversal',
+
+'S',
+
+'Expr', 'AtomicExpr', 'UnevaluatedExpr',
+
+'Symbol', 'Wild', 'Dummy', 'symbols', 'var',
+
+'Number', 'Float', 'Rational', 'Integer', 'NumberSymbol',
+'RealNumber', 'igcd', 'ilcm', 'seterr', 'E', 'I', 'nan', 'oo', 'pi', 'zoo',
+'AlgebraicNumber', 'comp', 'mod_inverse',
+
+'Pow', 'integer_nthroot', 'integer_log',
+
+'Mul', 'prod',
+
+'Add',
+
+'Mod',
+
+'Rel', 'Eq', 'Ne', 'Lt', 'Le', 'Gt', 'Ge',
+'Equality', 'GreaterThan', 'LessThan', 'Unequality', 'StrictGreaterThan',
+'StrictLessThan',
+
+'vectorize',
+
+'Lambda', 'WildFunction', 'Derivative', 'diff', 'FunctionClass',
+'Function', 'Subs', 'expand', 'PoleError', 'count_ops',
+'expand_mul', 'expand_log', 'expand_func',
+'expand_trig', 'expand_complex', 'expand_multinomial', 'nfloat',
+'expand_power_base', 'expand_power_exp', 'arity',
+
+'evalf', 'PrecisionExhausted', 'N',
+
+'Tuple', 'Dict',
+
+'gcd_terms', 'factor_terms', 'factor_nc',
+
+'evaluate',
+
```

Comment:
Lots of blank lines.

Reply:
Yes, I kept it in the same style as for `calculus` where each submodule/file has its own section of the list. Easy to change though. (I didn't check any other file, so no idea if it is consistent.)

https://github.com/sympy/sympy/blob/fb98dbb5a50dc13489b365ea6558e1efc5e62377/sympy/calculus/__init__.py#L1-L20

After checking a bit more now, there are not that many module with an `__all__` list... I noticed though that printing use a slightly different approach:

https://github.com/sympy/sympy/blob/master/sympy/printing/__init__.py